src/main/java/thewaypointers/trafficsimulator/TrafficSimulatorManager.java

```diff
@@ -1,76 +0,0 @@
-package thewaypointers.trafficsimulator;
-
-import thewaypointers.trafficsimulator.common.IStateProvider;
-import thewaypointers.trafficsimulator.common.WorldStateDTO;
-import thewaypointers.trafficsimulator.common.helpers.JunctionTestProvider;
-import thewaypointers.trafficsimulator.gui.MainFrame;
-
-import javax.swing.*;
-
-public class TrafficSimulatorManager extends Thread {
-
-    public static final float VEHICLE_MOVEMENT_SPEED = 2;
-    public static final float STATES_PER_SECOND = 1;
-    private static MainFrame mainFrame;
-    private static IStateProvider simulation;
-    private WorldStateDTO worldState;
-    private boolean stop = true;
-    private boolean isSleep = true;
-
-
-
-    private JButton start_pauseButton;
-
-    public TrafficSimulatorManager() {
-    }
-
-
-    public static void setSimulation(IStateProvider simulation) {
-        TrafficSimulatorManager.simulation = simulation;
-    }
-
-    public static void setMainFrame(MainFrame mainFrame) {
-        TrafficSimulatorManager.mainFrame = mainFrame;
-    }
-
-    public void setStart_pauseButton(JButton start_pauseButton) {
-        this.start_pauseButton = start_pauseButton;
-    }
-    public TrafficSimulatorManager(JButton start_pause){
-        start_pauseButton = start_pause;
-    }
-
-
-
-    public synchronized void changeState() {
-        isSleep = !isSleep;
-        start_pauseButton.setText("Start");
-    }
-
-    public synchronized void stopState() {
-        stop = !stop;
-        simulation = new JunctionTestProvider();
-        mainFrame.setVisible(false);
-        mainFrame = new MainFrame(simulation.getNextState(0),new TrafficSimulatorManager());
-    }
```

Comment:
There was a lot of things wrong in this method:
- Why did it flip the `stop` variable? We just want to stop when clearing, so `stop=true;` would be more appropriate. Besides, this is the same as pausing, so a separate variable is not needed.
- It had creation of new `JunctionTestProvider` hardcoded for clearing, which makes no sense because we want this to use with other providers as well. It would be best if the providers had a method for clearing, but it would take quite some time to implement it for `Simulation`, so for now creating a new provider should suffice, just not hardcoded. In my changes, this class takes a state provider supplier function that is capable of creating a new provider.
- Last thing is it basically hiding the current GUI and thread and creating a new one on top of it. There is no need to create everything from scratch, the GUI is built so when input changes, the GUI also changes. This would also make us lose current settings that user set in the control panel. Even if you wanted to recreate everything, you don't close the old copies. You don't stop the old thread, which means it's still running, and even though the `while(true)` loop doesn't execute its body, it's still spinning and taking CPU. Clicking Clear several times basically locks my computer up because of all the threads running.